### PR TITLE
Xboxkrnl additions

### PR DESCRIPTION
**Pull request by thrimbor**
I'm trying to clean up my branches and noticed that I didn't upstream those changes yet. They're only documentation changes and added NTSTATUS-constants (I believe they come from my work on exceptions), so no behavioral changes are expected.

### CODE REVIEW

File: lib/xboxkrnl/xboxkrnl.h [hunk 1]
```diff
@@ -4001,6 +4023,12 @@ XBAPI VOID DECLSPEC_NORETURN NTAPI HalReturnToFirmware
     IN FIRMWARE_REENTRY Routine
 );
 
+/**
+ * Registers or unregisters a shutdown notification which contains a function
+ * pointer that will be called on shutdown.
```
Comment (JayFoxRox) [inline lib/xboxkrnl/xboxkrnl.h]: nit: This is the only description comment which ends in a period.

File: lib/xboxkrnl/xboxkrnl.h [hunk 2]
```diff
@@ -3725,6 +3743,10 @@ XBAPI NTSTATUS NTAPI KeDelayExecutionThread
     IN PLARGE_INTEGER Interval
 );
 
+/**
+ * Connects an interrupt object, allowing it to receive interrupts
+ * @return FALSE if the interrupt is already connected or cannot be connected, TRUE if it was conected successfully.
```
Comment (JayFoxRox) [inline lib/xboxkrnl/xboxkrnl.h]: nit: This is the only parameter comment which ends in a period.

File: lib/xboxkrnl/xboxkrnl.h (hunk 1 above)
Comment (thrimbor) [reply lib/xboxkrnl/xboxkrnl.h]: Actually I seem to have used periods a few times in the past, but the non-period variant is more common. Changed.

File: lib/xboxkrnl/xboxkrnl.h (hunk 2 above)
Comment (thrimbor) [reply lib/xboxkrnl/xboxkrnl.h]: Same as above, changed.

**Pull request merged**